wallet-test/src/test/scala/org/bitcoins/wallet/models/UTXOSpendingInfoDAOTest.scala

```diff
@@ -0,0 +1,27 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.currency.Bitcoins
+import org.bitcoins.core.protocol.transaction.{
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.wallet.util.{BitcoinSWalletTest, WalletTestUtil}
+
+class UTXOSpendingInfoDAOTest extends BitcoinSWalletTest {
+  behavior of "UTXOSpendingInfoDAO"
+
+  it should "insert a UTXO and read it" in {
+    for {
+      created <- {
+        val outpoint = TransactionOutPoint(WalletTestUtil.sampleTxid,
+                                           WalletTestUtil.sampleVout)
+        val output = TransactionOutput(Bitcoins.one, WalletTestUtil.sampleSPK)
+        val privkeyPath = WalletTestUtil.sampleBip44Path
+        val utxo =
+          UTXOSpendingInfoDb(None, outpoint, output, privkeyPath, None, None)
+        utxoDAO.create(utxo)
+      }
+      read <- utxoDAO.read(created.id.get)
+    } yield assert(read.contains(created))
```

Comment:
I think this is the kind of situation where I prefer reading futures

Reply:
Sorry, what do you mean?

Reply:
I think this would read nicer as
```
val outpoint = TransactionOutPoint(WalletTestUtil.sampleTxid, WalletTestUtil.sampleVout)
val output = TransactionOutput(Bitcoins.one, WalletTestUtil.sampleSPK)
val privkeyPath = WalletTestUtil.sampleBip44Path
val utxo = UTXOSpendingInfoDb(id = None,
                              outPoint = outpoint,
                              output = output,
                              privKeyPath = privkeyPath,
                              redeemScriptOpt = None, // todo test this properly
                              scriptWitnessOpt = None) // todo test this properly

val createdF = utxoDAO.create(utxo)
val readF = createdF.flatMap(utxoDAO.read(_.id.get))

readF.map(assert(_.contains(created)))
```
or something closer to this